packages/features/ee/payments/components/Payment.tsx

```diff
@@ -47,7 +48,7 @@ type States =
 export default function PaymentComponent(props: Props) {
   const { t, i18n } = useLocale();
   const router = useRouter();
-  const { email, name, date } = router.query;
+  const { bookingUid } = router.query;
```

Comment:
`bookingUid` is being read from router.query here.
See this loom where I try to make the payment and reach a page with empty uid

Reply:
Sorry I misunderstood your previous comment. I added a new commit where the bookingUid comes from the props instead of the router.query. 

I'm not able to test this locally as I'm missing a key, will look into that a bit later. Does this fix your issue? @hariombalhara